lib/Providers/Qr/QRicketProvider.php

```diff
@@ -35,7 +35,7 @@ public function getQRCodeImage(string $qrtext, int $size): string
     public function getUrl(string $qrtext, int $size): string
     {
         return 'http://qrickit.com/api/qr'
-            . '?qrsize=' . (string) $size
+            . '?qrsize=' . $size
```

Comment:
why did you remove the cast to string? `size` is `int` here, and in a concat, one needs to cast to string first. Anyway, using `sprintf` would be better, no?

Reply:
PHP has type juggling. If you concatenate anything with a string that is not a string type, PHP will automatically convert it. The type cast is absolutely unnecessary there. See: https://onlinephp.io/c/e7113
Eg.
```php
$hello = "Hello";
$hello123 = $hello . 123; // Hello123
```

Reply:
Don't get me wrong, I'm not saying it is *necessary*, but rather *improper*, and the whole point is to avoid as much as possible type juggling and make things as explicit as possible. And I know of at least one static analysis tool (maybe rector?) that gives an error when concat str/int is encountered. So yeah, use sprintf!

Reply:
I've just updated the `getUrl` implementations to use `sprintf` in https://github.com/RobThree/TwoFactorAuth/pull/108/commits/4d91d6ae27e3b23d0af3be550956b8248a57ac5b.
Let me know your thoughts!

Reply:
I think we lose a lot on readability here. You can leave it like that it's alright, it's not a big deal, but I'm wondering if using `http_build_query` from an array wouldn't be better in term of readablity of the code :thinking: As a bonus, we would have proper url encoding!

Reply:
Check https://github.com/RobThree/TwoFactorAuth/pull/108/commits/cf84371dad11591b77b14c5dea5ac7c52a288b22, I think it improves a lot in terms of readability. Could it be improved further?

Reply:
Oh yeah, definitely much better and it allows to remove the urlencode stuff! :+1: